system/jlib/jatomic.hpp

```diff
@@ -23,8 +23,15 @@
 #ifdef _WIN32
 inline static void spinPause() { YieldProcessor(); }
 #else
-#include "xmmintrin.h"
+# include "x86intrin.h"
+# if defined(_ARCH_X86_)
```

Comment:
I think it is compiling on OSX (though 6.0.0 branch fails to build on OSX for other reasons at the moment)